### PR TITLE
Use WORD_SIZE_IN_BITS for Word32 guard

### PR DESCRIPTION
**Pull request by hasufell**
This should also fix build on armv7 wrt #961

### CODE REVIEW

**Comment by adithyaov** [comment]
LGTM. @pranaysashank Should we add a CI testing this change (build on 32-bit machine)?

**Comment by harendra-kumar** [comment]
We should put this fix in 0.7.3 as well.

**Comment by hasufell** [comment]
I've backported this patch to 0.7.2 and used that to successfully build ghcup for arm: https://github.com/hasufell/streamly/commit/f921e78236c6b44a6673b5e1a1803d2e7102d806

https://gitlab.haskell.org/haskell/ghcup-hs/-/jobs/597353

**Comment by pranaysashank** [comment]
@hasufell incorporating this fix in 0.7.3, can you check if this branch https://github.com/composewell/streamly/tree/staging-v0.7.3 builds successfully?

**Comment by hasufell** [comment]
@pranaysashank build is running here https://gitlab.haskell.org/haskell/ghcup-hs/-/pipelines/32074

**Comment by hasufell** [comment]
Can this be merged?

File: src/Streamly/Internal/Data/Time/TimeSpec.hsc [hunk 1]
```diff
@@ -24,7 +26,7 @@ module Streamly.Internal.Data.Time.TimeSpec
 where
 
 import Data.Int (Int64)
-#if i386_HOST_ARCH
+#if (WORD_SIZE_IN_BITS == 32)
```
Comment (pranaysashank) [inline src/Streamly/Internal/Data/Time/TimeSpec.hsc:29]: Is this conditional necessary ? Given that staging-v0.7.3 branch built successfully on arm-v7 (and it doesn't have this flag) I am thinking this won't be required

Comment (pranaysashank) [reply src/Streamly/Internal/Data/Time/TimeSpec.hsc:29]: Will result in unused import warning, don't remember seeing it in staging-v0.7.3. branch build. I'll merge this.

**Review by pranaysashank — APPROVED** [review]

**Pull request merged**